form/form_test.go

```diff
@@ -147,9 +169,28 @@ func (s *formSuite) TestIOPrompter(c *gc.C) {
 	}
 }
 
+func (s *formSuite) TestIOPrompterWrtiteErrors(c *gc.C) {
+	prompter := form.IOPrompter{
+		In:  nil,
+		Out: errorWriter{},
+	}
+	err := prompter.ShowTitle("title")
+	c.Assert(err, gc.ErrorMatches, "cannot show title: test")
+	_, err = prompter.Prompt("attr", environschema.Attr{
+		Description: "A",
+	})
+	c.Assert(err, gc.ErrorMatches, "cannot write prompt: test")
+}
+
+type errorWriter struct{}
+
+func (w errorWriter) Write([]byte) (n int, err error) {
+	return 0, errgo.New("test")
+}
+
 var promptingFillerTests = []struct {
 	about         string
-	schema        environschema.Fields
+	form          form.Form
 	maxTries      int
 	responses     []response
 	expectPrompts []prompt
```

Comment:
Shouldn't we have an expectShowTitle field here too, and a test or two that uses that?